test/recipes/30-test_evp_data/evppkey_kdf_hkdf.txt

```diff
@@ -193,3 +195,26 @@ Ctrl.md = md:SHA1
 Ctrl.IKM = hexkey:0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c
 Ctrl.salt = salt:
 Output = 2c91117204d745f3500d636a62f64f0ab3bae548aa53d423b0d1f27ebba6f5e5673a081d70cce7acfc48
+
+Title = FIPS indicator tests
+
+# Test that the key whose length is shorter than 112 bits is rejected
+FIPSversion = >=3.4.0
+PKEYKDF = HKDF
+Ctrl.digest = digest:SHA1
+Ctrl.IKM = hexkey:0b0b0b0b0b0b0b0b0b0b0b
+Ctrl.salt = hexsalt:000102030405060708090a0b0c
+Ctrl.info = hexinfo:f0f1f2f3f4f5f6f7f8f9
+Result = KDF_DERIVE_ERROR
+
+# Test that the key whose length is shorter than 112 bits is reported as
+# unapproved
+FIPSversion = >=3.4.0
+PKEYKDF = HKDF
+Unapproved = 1
```

Comment:
This needs a change to evp_test.c since it uses "PKEYKDF" not "KDF". pkey_kdf_test_run() is not doing the unapproved check that kdf_test_run() does i.e. it doesnt call pkey_check_fips_approved()

Reply:
Hmm - once you do this you may find it fails since I think these pkey ones are the 'old' legacy API's that go thru a different path in the provider. So I think providers/implementations/exchange/kdf_exch.c is going to need to support the indicator changes also (It may just work automatically, but if it doesnt that is where the code is)

Reply:
Sorry that I forget to check PKEYKDF. I have added approved check for it in evp_test.

Since EVP_KDF is used as underlying implementation in the `kdf_exch.c` file, I think it works automatically. After testing, it works indeed.